packages/commonwealth/server/scripts/setupAppRoutes.ts

```diff
@@ -133,92 +133,96 @@ const setupAppRoutes = (
     renderWithMetaTags(res, title, description, author, image);
   });
 
-  const renderProposal = async (
+  const renderThread = async (
     scope: string,
-    proposalType: string,
-    proposalId: string,
+    threadId: string,
     res,
     chain?: ChainInstance
   ) => {
-    // Retrieve title, description, and author from the database
-    let title, description, author, image;
     chain = chain || (await models.Chain.findOne({ where: { id: scope } }));
 
-    if (proposalType === 'discussion' && proposalId !== null) {
-      // Retrieve discussions
-      const proposal = await models.Thread.findOne({
-        where: { id: proposalId },
-        include: [
-          {
-            model: models.Chain,
-          },
-          {
-            model: models.Address,
-            as: 'Address',
-          },
-        ],
-      });
+    // Retrieve discussions
+    const thread = await models.Thread.findOne({
+      where: { id: threadId },
+      include: [
+        {
+          model: models.Chain,
+        },
+        {
+          model: models.Address,
+          as: 'Address',
+        },
+      ],
+    });
 
-      title = proposal ? decodeTitle(proposal.title) : '';
+    const title = thread ? decodeTitle(thread.title) : '';
+    const description = thread ? thread.plaintext : '';
+    const image = chain
+      ? `https://commonwealth.im${chain.icon_url}`
+      : DEFAULT_COMMONWEALTH_LOGO;
 
-      description = proposal ? proposal.plaintext : '';
-      image = chain
-        ? `https://commonwealth.im${chain.icon_url}`
-        : DEFAULT_COMMONWEALTH_LOGO;
-      try {
-        const profile = await models.Profile.findOne({
-          where: { id: proposal.Address.id },
-        });
-        author = profile.profile_name;
-      } catch (e) {
-        author = '';
-      }
-    } else {
-      title = chain ? chain.name : 'Commonwealth';
-      description = '';
-      image = chain
-        ? `https://commonwealth.im${chain.icon_url}`
-        : DEFAULT_COMMONWEALTH_LOGO;
+    let author;
+    try {
+      const profile = await models.Profile.findOne({
+        where: { id: thread.Address.id },
+      });
+      author = profile.profile_name;
+    } catch (e) {
       author = '';
     }
+
+    renderWithMetaTags(res, title, description, author, image);
+  };
+
+  const renderProposal = async (
+    scope: string,
+    res,
+    chain?: ChainInstance
+  ) => {
+    // Retrieve title, description, and author from the database
+    chain = chain || (await models.Chain.findOne({ where: { id: scope } }));
+
+    const title = chain ? chain.name : 'Commonwealth';
```

Comment:
What will this look like as a link? Can you post a screenshot? It would be weird to have a proposal link that reads out as just e.g. "dYdX"...

Reply:
Is there any way we can surface a better / more detailed description of the title / content?

Reply:
I had changed this to match the title of the post. For example:
<img width="1456" alt="Screenshot 2023-04-26 at 2 09 36 PM" src="https://user-images.githubusercontent.com/14794654/234703320-beb02ef5-2b95-4ec2-92d9-769e199481a6.png">

Reply:
Ideally on the ogp card viewer, the title should be the title of the thread, and the content should be a slice of the thread body...

Reply:
Ok, if I am understanding your comment correctly, this is how it is currently implemented. My screenshot above was bad, but the meta tags have two attributes, name and content. The element looks like: `<meta name="title" content="...">.

Reply:
Can you post a screenshot of how it renders? There's a tool on the wiki: https://github.com/hicommonwealth/commonwealth/wiki/Link-Previews for https://cards-dev.twitter.com/validator

Reply:
So twitters validator has been discontinued. Using https://web-toolbox.dev/en/tools/ogp-checker looks like:
<img width="1085" alt="Screenshot 2023-04-28 at 7 46 56 AM" src="https://user-images.githubusercontent.com/14794654/235179480-022485f1-87c5-4a77-96c9-f3dda6e44159.png">

Reply:
That looks great for threads. Can you try a proposal? Then we can get this in.

Reply:
Also wait that image url is broken.... look closely at it.

Reply:
Good catch. Fixed:
<img width="1080" alt="Screenshot 2023-04-28 at 2 22 00 PM" src="https://user-images.githubusercontent.com/14794654/235256434-e476f559-e04f-4015-a96b-3f66d9c7aff4.png">

And proposals:
<img width="1094" alt="Screenshot 2023-04-28 at 2 46 40 PM" src="https://user-images.githubusercontent.com/14794654/235259683-c502e9db-a069-493f-a3e4-1a7feefcd69c.png">